### PR TITLE
enh: Migrate photos picker to use NcDialog

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/photos/issues/2287

This simplifies the code and also fixes the double scrollbar issue.

### CODE REVIEW

**Review by Pytal — COMMENTED** [review]
The left nav doesn't show up anymore 🤔

**Comment by susnux** [comment]
> The left nav doesn't show up anymore 🤔

I only enabled the navigation for > 1 entries. Reverted.

**Review by Pytal — COMMENTED** [review]
image
Padding needs adjustment

File: src/components/PhotosPicker.vue [hunk 1]
```diff
@@ -20,51 +21,26 @@
...
+			<ul>
+				<li v-for="month in monthsList" :key="month">
+					<NcButton :type="targetMonth === month ? 'secondary' : 'tertiary'"
+						:aria-label="t('photos', 'Jump to {date}', { date: dateMonthAndYear(month) })"
+						@click="targetMonth = month">
+						{{ dateMonthAndYear(month) }}
+					</NcButton>
+				</li>
+			</ul>
+		</template>
```
Comment (Pytal) [inline src/components/PhotosPicker.vue:40]: When there are many months and viewport is 320px, it does not scroll

We should use NcSelect when in mobile view like in nc/dialogs FilePicker

**Review by Pytal — COMMENTED** [review]
When there are many months

File: src/components/PhotosPicker.vue (hunk 1 above)
Comment (susnux) [reply src/components/PhotosPicker.vue]: Done :heavy_check_mark: 


localhost_8082_index php_apps_photos_albums_a

**Comment by susnux** [comment]
>  Padding needs adjustment

Done 

Screenshot_20240201_005028

File: src/components/PhotosPicker.vue [hunk 2]
```diff
@@ -20,51 +21,45 @@
...
+		:open="open"
+		out-transition
+		size="large"
+		@update:open="(open) => $emit('update:open', open)">
+		<!-- Navigation containing the months available -->
+		<template #navigation="{ isCollapsed }">
+			<!-- Mobile view -->
+			<NcSelect v-if="isCollapsed"
+				v-model="targetMonth"
+				:aria-label-combobox="t('photos', 'Jump to specific date in list')"
```
Comment (Pytal) [inline src/components/PhotosPicker.vue:35]: We should use the always visible `inputLabel` here

Comment (Pytal) [inline src/components/PhotosPicker.vue:35]: An `aria-label-listbox="t('photos', 'Dates')"` should be added so screen reader users have "Dates list" announced

File: src/components/PhotosPicker.vue [hunk 3]
```diff
@@ -20,51 +21,45 @@
...
-		</div>
-
-		<div class="photos-picker__actions">
+				<template #option="{ label }">
+					{{ dateMonthAndYear(label) }}
+				</template>
+			</NcSelect>
+
+			<!-- Default view -->
+			<ul v-else>
```
Comment (Pytal) [inline src/components/PhotosPicker.vue:48]:
```suggestion
			<ul v-else aria-label="t('photos', 'Dates')">
```
For screen reader users similar to above

**Review by JuliaKirschenheuter — APPROVED** [review]

**Comment by susnux** [comment]
@Pytal resolved your comments, please re-review :)

**Comment by szaimen** [comment]
Sorry, I don't feel knowledgeable enough to review this...

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
Code looks good, but you probably need to update the cypress tests.

**Review by emoral435 — APPROVED** [review]
PR looks fantastic! Reviewed, and tested on my local dev, everything works as intended. 

NcDialog does have this weird look when the close button is focused, though 🤔
image

The active border gets clipped, let me know if you want me to fix this or just think that it is not a high priority issue right now :)

**Comment by susnux** [comment]
> NcDialog does have this weird look when the close button is focused, though 🤔

This should be fixed on the nextcloud-vue-library

**Review by Pytal — CHANGES_REQUESTED** [review]
Tested with a lot of photos and the photo at the end is cut off 🤔
image

**Comment by Pytal** [comment]
Consider this check ✅ my approval in advance once the cut off issue is fixed :D

**Comment by Pytal** [comment]
Cut off issue is still present but just tested master and it also has the same issue so will cc @artonge for investigation at a later point

**Review by Pytal — APPROVED** [review]
Works well! Just have to update cypress as mentioned above

**Pull request merged**